src/libs/state.ts

```diff
@@ -4,12 +4,72 @@ import type { WorkspaceWrapper } from 'src/libs/workspace-utils';
 
 export const routeHandlersStore = atom<unknown[]>([]);
 
-export const authStore = atom<any>({
+export type TerraUser = {
+  token?: string | undefined;
+  scope?: string | undefined;
+  id?: string | undefined;
+  email?: string | undefined;
+  name?: string | undefined;
+  givenName?: string | undefined;
+  familyName?: string | undefined;
+  imageUrl?: string | undefined;
+  idp?: string | undefined;
+};
+
+export type TerraUserProfile = {
+  institute: string | undefined;
+  title: string | undefined;
+  department: string | undefined;
+  interestInTerra: string | undefined;
+};
+
+export type TokenMetadata = {
+  token: string | undefined; // do not log or send this to mixpanel
+  id: string | undefined;
+  createdAt: number;
+  expiresAt: number;
+  totalTokensUsedThisSession: number;
+  totalTokenLoadAttemptsThisSession: number;
+};
+
+export type AuthState = {
+  anonymousId: string | undefined;
+  authContext: any;
+  authTokenMetadata: TokenMetadata;
+  cookiesAccepted: boolean | undefined;
+  fenceStatus: {};
+  hasGcpBillingScopeThroughB2C: boolean | undefined;
+  isAzurePreviewUser?: boolean | undefined;
+  isSignedIn: boolean | undefined;
+  isTimeoutEnabled?: boolean | undefined;
+  nihStatus?: {
+    linkedNihUsername: string;
+    linkExpireTime: number;
+  };
+  oidcConfig: {
+    authorityEndpoint?: string;
+    clientId?: string;
+  };
+  // props in the TerraUserProfile are always present, but there may be more props
+  profile: TerraUserProfile & any;
+  refreshTokenMetadata: TokenMetadata;
+  registrationStatus: any;
```

Comment:
I'm pretty sure this can be better typed as a string.

```suggestion
  registrationStatus: string | undefined;
```

Also, I think there's an existing "enum" of these values:
https://github.com/DataBiosphere/terra-ui/blob/1a1543d96eef720b56a262ccd6f5758a8ed06e93/src/libs/state.ts#L33-L38